### PR TITLE
5605 sync contact form

### LINKED ISSUE #5605: Sync feedback_form
Sync feedback form as remote data via v6 sync

Should only need to sync _to_ central, doesn't need to sync back again

### PR DESCRIPTION
**Pull request by GeronimoJohn**
Fixes #5605 

# 👩🏻‍💻 What does this PR do?

Developed the Contact Form Translator, designed exclusively to push or sync data to the omSupply central system. This functionality does not require syncing data back to the remote server, as the ContactForm table is used solely for saving data and is not queried.

Screenshot 2024-12-19 at 10 14 19 AM
Screenshot 2024-12-19 at 10 26 46 AM

## 💌 Any notes for the reviewer?

# 🧪 Testing

- [ ] Ensure omSupply central server and remote server are running
- [ ] Navigate to help page
- [ ] Fill out Contact Us and click send
- [ ] Navigate to Sync page
- [ ] Click Sync Now and wait for sync to finish
- [ ] Check omSupply central server database if record has been transferred

# 📃 Documentation

- [x] **Part of an epic**: documentation will be completed for the feature as a whole
- [ ] **No documentation required**: no user facing changes or a bug fix which isn't a change in behaviour
- [ ] **These areas should be updated or checked**: 
  1.
  2.

### CODE REVIEW

**Comment by jmbrunskill** [comment]
@GeronimoJohn I recently added a new ChangeLogSyncStyle for these records that only need to transfer in one direction.

See `ChangeLogSyncStyle::RemoteToCentral` in `changelog.rs`
It might be good to switch to this one to help clarify that it's a one directional sync...

File: server/service/src/sync/translations/contact_form.rs [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+        Some(ChangelogTableName::ContactForm)
+    }
+
+    fn should_translate_to_sync_record(
+        &self,
+        row: &ChangelogRow,
+        r#type: &ToSyncRecordTranslationType,
+    ) -> bool {
+        match r#type {
+            ToSyncRecordTranslationType::PullFromOmSupplyCentral => {
```
Comment (jmbrunskill) [inline server/service/src/sync/translations/contact_form.rs:49]: If you only want it to be pushed, you can skip this case...

Comment (jmbrunskill) [reply server/service/src/sync/translations/contact_form.rs:49]: Although You'll need to remove it from the `pull_upsert_records` tests though I think if you do...

Comment (GeronimoJohn) [reply server/service/src/sync/translations/contact_form.rs:49]: Yessss, thanks for that James, we we're hoping that we can keep to just be a push! I'll update the test accordingly. But I think this solves our issue right @lache-melvin?

Comment (lache-melvin) [reply server/service/src/sync/translations/contact_form.rs:49]: I think we tried this John? The issue you were seeing was in the `push` step of pull and push test, it's currently looking for pull records!

We should fix this test (definitely don't want bugs in our tests 😅, especially not around sync!) but I think should be done in a separate issue. I think there might be a bit of clean up involved, shouldn't block/bloat this PR 🙏

Comment (GeronimoJohn) [reply server/service/src/sync/translations/contact_form.rs:49]: Yea after thinking about it yesterday we still have to sort that test out. Though I also realised that James left two separate comments haha. I thought they were one.

**Review by lache-melvin — APPROVED** [review]
@jmbrunskill that remote to central sync is a nice add! I am a little confused though.. pull from central is still being implemented in the translator for system log?

Working well thanks John! I'm happy to merge this as is, and we can clean up the pull/push situation after resolving the test bug, I'll add some more context to the issue you created - thank you!

File: server/repository/src/db_diesel/contact_form_row.rs [hunk 2]
```diff
@@ -37,8 +37,8 @@ pub struct ContactFormRow {
     pub reply_email: String,
     pub body: String,
     pub created_datetime: NaiveDateTime,
-    pub store_id: String,
     pub user_id: String,
+    pub store_id: String,
```
[the remark this answers is not in the document]
Comment (GeronimoJohn) [reply server/repository/src/db_diesel/contact_form_row.rs:41]: 🙃

**Comment by jmbrunskill** [comment]
> @jmbrunskill that remote to central sync is a nice add! I am a little confused though.. pull from central is still being implemented in the translator for system log?
> 
> Working well thanks John! I'm happy to merge this as is, and we can clean up the pull/push situation after resolving the test bug, I'll add some more context to the issue you created - thank you!

Yes, Sorry for adding confusion. And thanks @GeronimoJohn  for adding that issue for fixing this!

**Pull request merged**